### PR TITLE
apps/examples/testcase/ta_tc: Add BLE manager UTC application

### PR DESCRIPTION
**Pull request by hdotkashyap**
Unit Test Cases (positive & negative) for the APIs of BLE manager

Signed-off-by: Hrishikesh Kashyap <h1.kashyap@samsung.com>

### CODE REVIEW

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c [hunk 1]
```diff
@@ -0,0 +1,1172 @@
...
+    filter->raw_data_length = len;
+  }
+
+  filter->scan_duration = scan_duration;
+  filter->whitelist_enable = whitelist_enable;
+}
+
+static void utc_blemanager_init_p(void)
+{
+  TC_ASSERT_EQ("ble_manager_init", ble_manager_init(NULL), BLE_MANAGER_SUCCESS);//not server role
```
Comment (amogh-samsung) [inline apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c:169]: Please use tab spaces for indentation.

File: apps/examples/testcase/ta_tc/ble_manager/utc/Kconfig [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# For a description of the syntax of this configuration file,
+# see kconfig-language at https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt
+#
+
+config EXAMPLES_TESTCASE_BLE_MANAGER_UTC
+	bool "BLE Manager UTC TestCase Example"
+	select BLE_MANAGER
+	default n
+	---help---
+		Enable the BLE Manager TestCase Example
```
Comment (amogh-samsung) [inline apps/examples/testcase/ta_tc/ble_manager/utc/Kconfig:11]: Do we need a newline character at the end here?

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c [hunk 3]
```diff
@@ -0,0 +1,1172 @@
...
+static void ble_device_connected_cb(ble_client_ctx *ctx, ble_device_connected *dev)
+{
+  printf("client connected callback received\n");
+  g_client_connected = BEVT_CONNECTED;
+  UTC_FUNC_SIGNAL;
+  return;
+}
+
+static void ble_operation_notification_cb(ble_client_ctx *ctx, ble_attr_handle attr_handle, ble_data *read_result)
+{
```
[the remark this answers is not in the document]
Comment (hdotkashyap) [reply apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c:89]: Done

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c [hunk 4]
```diff
@@ -0,0 +1,1172 @@
...
+  printf("notification callback received\n");
+  g_client_noti = BEVT_NOTI;
+  UTC_FUNC_SIGNAL;
+  return;
+}
+
+static void ble_server_connected_cb(ble_conn_handle con_handle, ble_server_connection_type_e conn_type, uint8_t mac[BLE_BD_ADDR_MAX_LEN])
+{
+  printf("server connect callback received");
```
[the remark this answers is not in the document]
Comment (hdotkashyap) [reply apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c]: Done

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c (hunk 1 above)
Comment (hdotkashyap) [reply apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c]: Done

File: apps/examples/testcase/ta_tc/ble_manager/utc/Kconfig (hunk 2 above)
Comment (hdotkashyap) [reply apps/examples/testcase/ta_tc/ble_manager/utc/Kconfig]: Done

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c [hunk 5]
```diff
@@ -0,0 +1,1172 @@
...
+{
+	TC_ASSERT_EQ("ble_manager deinit", ble_manager_deinit(), BLE_MANAGER_INVALID_STATE);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_blemanager_mac_p(void)
+{
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(NULL), BLE_MANAGER_SUCCESS);
+	uint8_t mac[BLE_BD_ADDR_MAX_LEN];
+	TC_ASSERT_EQ("ble_manager_get_mac_addr", ble_manager_get_mac_addr(mac), BLE_MANAGER_SUCCESS);
```
Comment (jeongchanKim) [inline apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c:203]: Is it okay not to deinit the ble manager if failed to get the mac addr?
I think we need to use `TC_ASSERT_EQ_CLEANUP`.

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c [hunk 6]
```diff
@@ -0,0 +1,1172 @@
...
+	conn_info.slave_latency = 128;
+	conn_info.mtu = 240;
+	conn_info.scan_timeout = 1000;
+	conn_info.is_secured_connect = true;
+	TC_ASSERT_EQ("ble_client_connect", ble_client_connect(ctx, &conn_info), BLE_MANAGER_SUCCESS);
+	CONTROL_VVDRIVER(VBLE_CMD_GEN_EVT, LWNL_EVT_BLE_CLIENT_CONNECT, 0, 3);
+	UTC_FUNC_WAIT;
+	TC_ASSERT_EQ("ble_client_connected cb", g_client_connected, BEVT_CONNECTED);
+	TC_ASSERT_EQ("ble_manager_get_bonded_device", ble_manager_get_bonded_device(dev_list, &dev_count), BLE_MANAGER_SUCCESS);
+	printf("Bonded device count: %d\n", dev_count);
```
Comment (jeongchanKim) [inline apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c:237]: Is the `dev_count` needed to check?
If then, it is better to use `TC_ASSERT_EQ` macro.
If not, is this needed to print?

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c [hunk 7]
```diff
@@ -0,0 +1,1172 @@
...
+	conn_info.slave_latency = 128;
+	conn_info.mtu = 240;
+	conn_info.scan_timeout = 1000;
+	conn_info.is_secured_connect = true;
+	TC_ASSERT_EQ("ble_client_connect", ble_client_connect(ctx, &conn_info), BLE_MANAGER_SUCCESS);
+	CONTROL_VVDRIVER(VBLE_CMD_GEN_EVT, LWNL_EVT_BLE_CLIENT_CONNECT, 0, 3);
+	UTC_FUNC_WAIT;
+	TC_ASSERT_EQ("ble_client_connected cb", g_client_connected, BEVT_CONNECTED);
+	TC_ASSERT_EQ("ble_manager_conn_is_active", ble_manager_conn_is_any_active(&is_active), BLE_MANAGER_SUCCESS);
+	if(is_active == true) {
```
Comment (jeongchanKim) [inline apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c:409]: Please add space after `if`.

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c [hunk 8]
```diff
@@ -0,0 +1,1172 @@
...
+	true,
+	gatt_profile,
+	sizeof(gatt_profile) / sizeof(ble_server_gatt_t)
+};
+
+static void set_scan_filter(ble_scan_filter *filter, uint8_t *raw_data, uint8_t len, bool whitelist_enable, uint32_t scan_duration)
+{
+	memset(filter, 0, sizeof(ble_scan_filter));
+	if (raw_data != NULL && len > 0) {
+		memcpy(filter->raw_data, raw_data, len);
```
[the remark this answers is not in the document]
Comment (hdotkashyap) [reply apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c:159]: memset initialises all the members of **filter** which also includes **uint8_t raw_data[BLE_ADV_RAW_DATA_MAX_LEN]** .
User specified **raw_data** then gets copied there.

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c (hunk 7 above)
Comment (hdotkashyap) [reply apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c]: Done

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c (hunk 6 above)
Comment (hdotkashyap) [reply apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c]: Done

File: apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c (hunk 5 above)
Comment (hdotkashyap) [reply apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c]: Done

**Pull request closed without merging**